assimilation_code/modules/io/io_filenames_mod.f90

```diff
@@ -632,24 +640,39 @@ subroutine check_attributes(ncFile, filename, ncVarId, domid, varid)
 if ( get_has_missing_value(domid, varid) ) then
    select case (get_xtype(domid,varid))
       case (NF90_INT)
-         call get_FillValue(domid, varid, spvalINT)
-         call check_attribute_value_int(ncFile, filename, ncVarID, '_FillValue', spvalINT)
          call get_missing_value(domid, varid, spvalINT)
          call check_attribute_value_int(ncFile, filename, ncVarID, 'missing_value', spvalINT)
 
       case (NF90_FLOAT)
-         call get_FillValue(domid, varid, spvalR4)
-         call check_attribute_value_r4(ncFile, filename, ncVarID, '_FillValue', spvalR4)
          call get_missing_value(domid, varid, spvalR4)
          call check_attribute_value_r4(ncFile, filename, ncVarID, 'missing_value', spvalR4)
 
       case (NF90_DOUBLE)
-         call get_FillValue(domid, varid, spvalR8)
-         call check_attribute_value_r8(ncFile, filename, ncVarID, '_FillValue', spvalR8)
          call get_missing_value(domid, varid, spvalR8)
          call check_attribute_value_r8(ncFile, filename, ncVarID, 'missing_value', spvalR8)
 
       case default
+         !>@todo FIXME report the variable with the unsupported xtype
+         call error_handler(E_ERR, 'check_attributes', 'unknown xtype', source)
+   end select
+endif
+
+if ( get_has_FillValue(   domid, varid) ) then
+   select case (get_xtype(domid, varid))
+      case (NF90_INT)
+         call get_FillValue(domid, varid, spvalINT)
+         call check_attribute_value_int(ncFile, filename, ncVarID, 'FillValue', spvalINT)
```

Comment:
why is this 'FillValue' and not '_FillValue'?

Reply:
There is no error check in check_attribute_value if nf90_get_att returns NF90_NOERR, so I think this code has merrily continues because there is no attribute called 'FillValue'

Reply:
fixed this in 72c1d2de08fc16f70fcfc816fd6f19aa6f05fc8c